packages/liveblocks-react/src/room.tsx

```diff
@@ -1413,10 +1413,9 @@ export function createRoomContext<
                     ...state.threads,
                     [threadId]: {
                       ...existingThread,
-                      metadata: {
-                        ...existingThread.metadata,
-                        ...metadata,
-                      },
+                      metadata: metadata as [TThreadMetadata] extends [never]
+                        ? Record<string, never>
```

Comment:
This is the fix!

The server responds with the total metadata. We used to merge the metadata received from server with the existing metadata, so there was a possibility that stale properties were still stored until a webhook message cleared them.